src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -126,6 +126,45 @@ trait WorkspaceComponent {
       DBIO.seq(insertActions.toSeq:_*)
     }
 
+    private def upsertAttributes(workspace: Workspace) = {
+      val workspaceId = UUID.fromString(workspace.workspaceId)
+
+      def createWorkspaceAttributeTempTable() = {
+        sql"""create temporary table if not exists WORKSPACE_ATTRIBUTE_TEMP (id bigint(20) unsigned NOT NULL AUTO_INCREMENT primary key, name text NOT NULL, value_string text, value_number double DEFAULT NULL, value_boolean bit(1) DEFAULT NULL, value_entity_ref bigint(20) unsigned DEFAULT NULL, list_index int(11) DEFAULT NULL, owner_id binary(16) NOT NULL)""".as[Int]
```

Comment:
we're using temp tables because it allows nearly all of the logic for deleting for deleting/inserting/updating attributes to be done on the database side. this is better because it will result in far fewer queries and code that is much less complex. we don't have to fully load any of the attributes and their references in order to know that they have changed